include/caffe/net.hpp

```diff
@@ -25,6 +25,7 @@ class Net {
  public:
   explicit Net(const NetParameter& param, const Net* root_net = NULL);
   explicit Net(const string& param_file, Phase phase,
+      const int level = 0, const vector<string>* stages = NULL,
```

Comment:
I agree, that seems like the safer choice. I wonder why no tests failed? Seems like a test hole.